File: 70.js
function isValid(s) {
    let stack = [];
    let map = {")": "(", "]": "[", "}": "{"};

    for (let c of s) {
        if (c === "(" || c === "[" || c === "{") {
            stack.push(c);
        } else {
            if (!stack.length || stack[stack.length - 1] !== map[c])
                return false;
            stack.pop();
        }
    }
    return stack.length === 0;
}

// Driver Code
console.log(isValid("()[]{}")); // true
console.log(isValid("(]"));     // false

File: 70.py
def isValid(s):
    stack = []
    mapping = {')':'(', ']':'[', '}':'{'}

    for c in s:
        if c in mapping.values():
            stack.append(c)
        else:
            if not stack or stack[-1] != mapping[c]:
                return False
            stack.pop()
    return len(stack) == 0


# Driver Code
print(isValid("()[]{}"))  # True
print(isValid("(]"))      # False
